### PR TITLE
#1612: [Refactoring] Templates

### LINKED ISSUE #1612: [Refactoring] Templates
Related to https://github.com/magento/adobe-stock-integration/issues/1578

As per code review we need to avoid the use of legacy syntax. More info can be found here https://devdocs.magento.com/guides/v2.4/ui_comp_guide/concepts/magento-bindings.html

Files to refactor:
- `MediaGalleryUi/view/adminhtml/templates/image_details.phtml:30`
- `MediaGalleryUi/view/adminhtml/templates/image_details_standalone.phtml:28`
- `MediaGalleryUi/view/adminhtml/templates/image_edit_details.phtml:28`
- `MediaGalleryUi/view/adminhtml/templates/image_edit_details_standalone.phtml:34`
- `MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html:24`
- `MediaGalleryUi/view/adminhtml/web/template/grid/columns/image/actions.html:8`
- `MediaGalleryUi/view/adminhtml/web/template/grid/massactions/count.html:8`
- `MediaGalleryUi/view/adminhtml/web/template/image-uploader.html:14` - need to fix that
- `MediaGalleryUi/view/adminhtml/web/template/image/actions.html:7`
- `MediaGalleryUi/view/adminhtml/web/template/image/image-details.html:39`
- `MediaGalleryUi/view/adminhtml/web/template/image/image-edit.html:13`
- `MediaGalleryUi/view/adminhtml/web/template/grid/filters/elements/ui-select.html:34`

### PR DESCRIPTION
**Pull request by jmonteros422**
#1612 

### Description (*)

Related to https://github.com/magento/adobe-stock-integration/issues/1578

As per code review we need to avoid the use of legacy syntax. More info can be found here https://devdocs.magento.com/guides/v2.4/ui_comp_guide/concepts/magento-bindings.html

Files to refactor:
- `MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html:24`
- `MediaGalleryUi/view/adminhtml/web/template/grid/columns/image/actions.html:8`
- `MediaGalleryUi/view/adminhtml/web/template/grid/massactions/count.html:8`
- `MediaGalleryUi/view/adminhtml/web/template/image-uploader.html:14` - need to fix that
- `MediaGalleryUi/view/adminhtml/web/template/image/actions.html:7`
- `MediaGalleryUi/view/adminhtml/web/template/image/image-details.html:39`
- `MediaGalleryUi/view/adminhtml/web/template/image/image-edit.html:13`
- `MediaGalleryUi/view/adminhtml/web/template/grid/filters/elements/ui-select.html:34`

### CODE REVIEW

File: MediaGalleryUi/view/adminhtml/web/template/image-uploader.html [hunk 1]
```diff
@@ -11,8 +11,7 @@
     </form>
     <div data-role="spinner" class="admin__data-grid-loading-mask" data-bind="visible: loader">
         <div class="spinner">
-            <span></span><span></span><span></span><span></span>
-            <span></span><span></span><span></span><span></span>
+            <span repeat="8"/>
```
Comment (lenaorobei) [inline MediaGalleryUi/view/adminhtml/web/template/image-uploader.html:14]: Is it possible to get rid of this at all?
Maybe, use styles to properly display spinner?

Comment (jmonteros422) [reply MediaGalleryUi/view/adminhtml/web/template/image-uploader.html:14]: Hi @lenaorobei ,

Thank you for checking this out. I believe this is how the core modules implements the spinner in the admin area. 
see samples: 
app/code/Magento/Catalog/view/adminhtml/templates/product/edit/tabs.phtml:88
app/code/Magento/Ui/view/base/web/templates/dynamic-rows/templates/grid.html:44

Let me know your thoughts on this. 

Thanks!

**Comment by jmonteros422** [comment]
@magento run all tests

File: MediaGalleryUi/view/adminhtml/web/template/grid/columns/image/actions.html [hunk 2]
```diff
@@ -5,9 +5,9 @@
  */
 -->
 
-<!-- ko  foreach: { data: actionsList, as: 'action' } -->
+<each args="{ data: actionsList, as: 'action' }">
     <li>
         <a class="action-menu-item" href=""
            data-bind="text: action.title, click: $parent[action.handler].bind($parent, $row()), 'data-action': 'item-' + action.name" />
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/template/grid/columns/image/actions.html]: Can you please check if remaining bindings can also be updated?
```suggestion
           text="action.title" click="$parent[action.handler].bind($parent, $row())" attr="'data-action': 'item-' + action.name" />
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @jmonteros422 thanks for the pull request, please see my comments

File: MediaGalleryUi/view/adminhtml/web/template/grid/columns/image/actions.html (hunk 2 above)
Comment (jmonteros422) [reply MediaGalleryUi/view/adminhtml/web/template/grid/columns/image/actions.html]: Thanks for the review @sivaschenko ! Bindings is now updated. Although this particular template seems to be overriden by AdobeStockImageAdminUi/view/adminhtml/web/template/mediaGallery/grid/columns/image/licenseActions.html , I have updated it as well.

**Comment by jmonteros422** [comment]
@magento run all tests

File: MediaGalleryUi/view/adminhtml/web/template/image/actions.html [hunk 3]
```diff
@@ -4,14 +4,9 @@
...
-            data-bind="click: $parent[action.handler].bind($parent),
-                       attr: {
-                            class: action.classes,
-                            id: 'image-details-action-' + action.name,
-                            title: $t(action.title)
-                       }">
+<each args="{ data: actionsList, as: 'action' }">
+    <button type="button" click="$parent[action.handler].bind($parent)"
+            attr="{class: action.classes, id: 'image-details-action-' + action.name, title: $t(action.title)}">
         <span data-bind="i18n: action.title"></span>
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/template/image/actions.html]: @jmonteros422 can you please search the `data-bind` usages and update them all at once?

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @jmonteros422 , thanks for the updates! Please see my comment

**Comment by jmonteros422** [comment]
@magento run all tests

File: AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html [hunk 4]
```diff
@@ -30,18 +30,18 @@
...
             <span></span><span></span><span></span><span></span>
         </div>
         <div if="showSeriesTab($row())" attr="'id': 'series_content_' + $row().id">
             <!-- ko foreach: getSeries($row()) -->
-            <div class="thumbnail" data-bind="click: function(){ $parent.switchImagePreviewToSeriesImage($data) }">
+            <div class="thumbnail" click="function(){ $parent.switchImagePreviewToSeriesImage($data) }">
                 <img attr="src: thumbnail_url, alt: title">
             </div>
             <!-- /ko -->
             <!-- ko if: canShowMoreSeriesImages($row()) -->
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html]:
```suggestion
            <if args="canShowMoreSeriesImages($row())">
```

File: AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/actions.html [hunk 5]
```diff
@@ -9,27 +9,25 @@
         <div class="messages">
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/actions.html:9]: <each args="getMessages()"

File: AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html [hunk 6]
```diff
@@ -30,18 +30,18 @@
 
     <div id="adobe-stock-tabs-content"
          class="adobe-stock-related-images-tab-content">
-        <div class="spinner related-loader" data-bind="visible: loader">
+        <div class="spinner related-loader" visible="loader">
             <span></span><span></span><span></span><span></span>
             <span></span><span></span><span></span><span></span>
         </div>
         <div if="showSeriesTab($row())" attr="'id': 'series_content_' + $row().id">
             <!-- ko foreach: getSeries($row()) -->
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html]:
```suggestion
            <eacg args="getSeries($row())">
```

File: AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html [hunk 7]
```diff
@@ -55,12 +55,12 @@
         </div>
         <div if="showModelTab($row())" attr="'id': 'model_content_' + $row().id">
             <!-- ko foreach: getModel($row()) -->
-            <div class="thumbnail" data-bind="click: function(){ $parent.switchImagePreviewToModelImage($data) }">
+            <div class="thumbnail" click="function(){ $parent.switchImagePreviewToModelImage($data) }">
                 <img attr="src: thumbnail_url, alt: title">
             </div>
             <!-- /ko -->
             <!-- ko if: canShowMoreModelImages($row()) -->
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html]:
```suggestion
            <if args="canShowMoreModelImages($row())">
```

File: AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html [hunk 8]
```diff
@@ -55,12 +55,12 @@
         </div>
         <div if="showModelTab($row())" attr="'id': 'model_content_' + $row().id">
             <!-- ko foreach: getModel($row()) -->
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/template/grid/column/preview/related.html]:
```suggestion
            <each args="getModel($row())">
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @jmonteros422 please see some remaining knockout bindings that I can see in the PR (there might be more if you search for `<!-- ko` in the `.html` files)

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento run all tests

File: MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html [hunk 9]
```diff
@@ -20,19 +20,19 @@
...
-            <li class="name" data-ui-id="title" data-bind="text: $row().title"></li>
+            <li class="name" data-ui-id="title" text="$row().title"></li>
             <li class="source">
-                <img if="$row().source" class="adobe-stock-icon" data-bind="attr: { src: $row().source }" />
+                <img if="$row().source" class="adobe-stock-icon" attr="{ src: $row().source }"/>
             </li>
-            <li class="type" data-ui-id="content-type" data-bind="text: $row().content_type"></li> &#8226;
-            <li class="dimensions" data-ui-id="dimensions" data-bind="text: $row().width + 'x' + $row().height"></li>
+            <li class="type" data-ui-id="content-type" text="$row().content_type"></li> &#8226;
+            <li class="dimensions" data-ui-id="dimensions" text="text: $row().width + 'x' + $row().height"></li>
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html]:
```suggestion
            <li class="dimensions" data-ui-id="dimensions" text="$row().width + 'x' + $row().height"></li>
```

File: MediaGalleryUi/view/adminhtml/web/template/image/image-edit.html [hunk 10]
```diff
@@ -7,37 +7,37 @@
...
+                    <span translate="'Title'"></span>
                 </label>
                 <div class="admin__field-control">
                     <input type="text" id="title" data-ui-id="title" name="title" placeholder="Title"
-                           class="admin__control-text required-entry minimum-length-1 maximum-length-128"
-                           data-bind="value: image().title, event: {keypress: handleEnterKey}"
-                           data-validate="{'required':true,'validate-image-title':true, 'validate-length':true}"/>
+                           class="admin__control-text required-entry minimum-length-1 maximum-length-120"
+                           ko-value="image().title" event="{keypress: handleEnterKey}"
+                           data-validate="{'required':true,'validate-alphanum-with-spaces':true, 'validate-length':true}"/>
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/template/image/image-edit.html]:
```suggestion
                           data-validate="{'required':true, 'validate-image-title':true, 'validate-length':true}"/>
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@jmonteros422 please pay more attention during testing the changes, that's a screenshot straight from the media gallery grid:
image

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**